### ISSUE TITLE
[MSVC] Build failure with error C2131 error

### ISSUE THREAD

**Issue by xkszltl** [facebook/rocksdb#4917]
Get the following error recently on Win10 with VS 2017

```
[1/12] Building CXX object CMakeFiles\rocksdb.dir\tools\db_bench_tool.cc.obj
FAILED: CMakeFiles/rocksdb.dir/tools/db_bench_tool.cc.obj
C:\PROGRA~2\MICROS~4\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\cl.exe  /nologo /TP -DGFLAGS=gflags -DHAVE_PCLMUL -DHAVE_SSE42 -DNOMINMAX -DOS_WIN -DROCKSDB_SUPPORT_THREAD_LOCAL -DSNAPPY -DWIN32 -DWIN64 -DXPRESS -DZLIB -D_MBCS -I"C:\P
rogram Files\GFlags\include" -I"C:\Program Files\Snappy\include" -I"C:\Program Files\zlib\include" -I..\ -I..\include -I..\third-party\gtest-1.7.0\fused-src /EHsc /GL /MP /Zi /nologo /EHsc /GS /Gd /GR /GF /fp:precise /Zc:wchar_t /Zc:forScope /err
orReport:queue /FC /d2Zi+ /W4 /wd4127 /wd4800 /wd4996 /wd4351 /wd4100 /wd4204 /wd4324 /arch:AVX2 /MD /O2 /Ob2 /DNDEBUG /GR- /Oxt /Zp8 /Gm- /Gy /MD   /Fd./rocksdb.pdb /showIncludes /FoCMakeFiles\rocksdb.dir\tools\db_bench_tool.cc.obj /FdCMakeFiles
\rocksdb.dir\rocksdb.pdb /FS -c ..\tools\db_bench_tool.cc
cl : Command line warning D9025 : overriding '/GR' with '/GR-'
cl : Command line warning D9025 : overriding '/Fd./rocksdb.pdb' with '/FdCMakeFiles\rocksdb.dir\rocksdb.pdb'
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4675): error C2131: expression did not evaluate to a constant
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4675): note: failure was caused by a read of a variable outside its lifetime
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4675): note: see usage of 'value_max'
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4692): warning C4244: 'argument': conversion from 'double' to 'int64_t', possible loss of data
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4695): warning C4244: 'argument': conversion from 'double' to 'int64_t', possible loss of data
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4731): warning C4244: 'argument': conversion from 'double' to 'int64_t', possible loss of data
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4733): warning C4244: 'argument': conversion from 'double' to 'int64_t', possible loss of data
[2/12] Building CXX object CMakeFiles\rocksdb-shared.dir\tools\db_bench_tool.cc.obj
FAILED: CMakeFiles/rocksdb-shared.dir/tools/db_bench_tool.cc.obj
C:\PROGRA~2\MICROS~4\2017\ENTERP~1\VC\Tools\MSVC\1416~1.270\bin\Hostx64\x64\cl.exe  /nologo /TP -DGFLAGS=gflags -DHAVE_PCLMUL -DHAVE_SSE42 -DNOMINMAX -DOS_WIN -DROCKSDB_DLL -DROCKSDB_LIBRARY_EXPORTS -DROCKSDB_SUPPORT_THREAD_LOCAL -DSNAPPY -DWIN32
 -DWIN64 -DXPRESS -DZLIB -D_MBCS -Drocksdb_shared_EXPORTS -I"C:\Program Files\GFlags\include" -I"C:\Program Files\Snappy\include" -I"C:\Program Files\zlib\include" -I..\ -I..\include -I..\third-party\gtest-1.7.0\fused-src /EHsc /GL /MP /Zi /nolog
o /EHsc /GS /Gd /GR /GF /fp:precise /Zc:wchar_t /Zc:forScope /errorReport:queue /FC /d2Zi+ /W4 /wd4127 /wd4800 /wd4996 /wd4351 /wd4100 /wd4204 /wd4324 /arch:AVX2 /MD /O2 /Ob2 /DNDEBUG /GR- /Oxt /Zp8 /Gm- /Gy /MD   /Fd./rocksdb-shared.pdb /showInc
ludes /FoCMakeFiles\rocksdb-shared.dir\tools\db_bench_tool.cc.obj /FdCMakeFiles\rocksdb-shared.dir\ /FS -c ..\tools\db_bench_tool.cc
cl : Command line warning D9025 : overriding '/GR' with '/GR-'
cl : Command line warning D9025 : overriding '/Fd./rocksdb-shared.pdb' with '/FdCMakeFiles\rocksdb-shared.dir\'
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4675): error C2131: expression did not evaluate to a constant
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4675): note: failure was caused by a read of a variable outside its lifetime
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4675): note: see usage of 'value_max'
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4692): warning C4244: 'argument': conversion from 'double' to 'int64_t', possible loss of data
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4695): warning C4244: 'argument': conversion from 'double' to 'int64_t', possible loss of data
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4731): warning C4244: 'argument': conversion from 'double' to 'int64_t', possible loss of data
c:\users\tolia\appdata\local\temp\rocksdb\tools\db_bench_tool.cc(4733): warning C4244: 'argument': conversion from 'double' to 'int64_t', possible loss of data
```

**Comment by xkszltl** [comment]
https://github.com/facebook/rocksdb/blame/e07aa8669daa887a7599d4588e15529d9b087c70/tools/db_bench_tool.cc#L4675

This line uses dynamic array, which is not standard C++.

**Comment by zhichao-cao** [comment]
> [rocksdb/tools/db_bench_tool.cc](https://github.com/facebook/rocksdb/blob/e07aa8669daa887a7599d4588e15529d9b087c70/tools/db_bench_tool.cc#L4675)
> 
> Line 4675 in [e07aa86](/facebook/rocksdb/commit/e07aa8669daa887a7599d4588e15529d9b087c70)
> 
>  char value_buffer[2 * value_max]; 
> This line from #4788 uses dynamic array, which is not standard C++.

Fixing it now.

**Comment by sagar0** [comment]
Fixed in #4918 .
